src/main/java/io/cdap/plugin/snowflake/common/util/SchemaHelper.java

```diff
@@ -95,6 +120,26 @@ private static Schema getParsedSchema(String schema) {
     }
   }
 
+  private static Schema getSchema(SnowflakeAccessor snowflakeAccessor, @Nullable String tableName,
+                                  @Nullable String importQuery) {
+    try {
+      List<SnowflakeFieldDescriptor> result;
+      if (!Strings.isNullOrEmpty(tableName)) {
+        result = snowflakeAccessor.describeTable(snowflakeAccessor.getSchema(), tableName);
+      } else {
+        result = snowflakeAccessor.describeQuery(importQuery);
+      }
+      List<Schema.Field> fields = result.stream()
+        .map(fieldDescriptor -> Schema.Field.of(fieldDescriptor.getName(), getSchema(fieldDescriptor)))
+        .collect(Collectors.toList());
+      return Schema.recordOf("data", fields);
+    } catch (SQLException e) {
+      throw new SchemaParseException(e);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
```

Comment:
any exception should be returned as `SchemaParseException` here

Reply:
done bf54c0020a6276a88d2cdafee7f88cd93b3b7904